esp-hal/src/gpio/mod.rs

```diff
@@ -1426,6 +1470,14 @@ impl<'d> Input<'d> {
     ///
     /// The output signal can be passed to peripherals in place of an output
     /// pin.
+    /// ```rust, no_run
```

Comment:
I'm not so sure about these GPIO doc comments, they seem to, on some level repeat themselves. Perhaps we should just document `split`, and leave the rest? Or maybe we can reference split in the other API items.

Not that stress about this at the moment though, as these are unstable APIs :)